### PR TITLE
EVM: fix EOF fixtures

### PR DESCRIPTION
**Pull request by jochem-brouwer**
This PR fixes the failing EOF fixtures as of: https://github.com/ethereum/execution-spec-tests/releases/tag/eip7692%40v2.1.0

To run: download the fixtures and extract those in `./packages/ethereum-tests`. (This will extract a `fixtures` folder).

To run, `cd` into the vm package and:

`npm run test:blockchain -- --fork=Osaka --dir=../fixtures/blockchain_tests` (blockchain tests)

`npm run test:state -- --fork=Osaka --dir=../fixtures/state_tests` (state tests)

NOTE: the blockchain-tests currently need to disable the gas limit validation (`blockchain/src/blockchain.ts` line 554 disable `header.validateGasLimit(parentHeader)` and the base fee calculation (`//throw new Error(`Invalid block: base fee not correct ${header.errorStr()}`)`, this is due to tests being wrongly filled with wrong gas limit / base fee data when compared to the genesis block. Tests pass if these are disabled).

To run the eof container tests:

Rename `evm/test/eips/eof-header-validation.ts` to something like `eof-header-validation.spec.ts` and ensure that `testDir` (line 14) of the file points to the `eof_tests` of the fixtures you unpacked. Run it with `npx vitest run ./test/eips/eof-header-validation.spec.ts`. All pass.

### CODE REVIEW

**Comment by holgerd77** [comment]
Can this be closed?

**Comment by jochem-brouwer** [comment]
No, actually I should import the changes in this branch for fixes for the EOF test fixtures (or use this PR for it).

**Comment by jochem-brouwer** [comment]
This currently passes the blockchain/state tests, what is left to do is to verify the container validator tests also.

**Comment by jochem-brouwer** [comment]
Problem with tests related to Prague, not Osaka: EXTCODEHASH on an empty account (with 7702 activated), test:

`npm run test:blockchain -- --fork=Osaka --dir=../fixtures/fixtures/blockchain_tests/osaka/eofwrap/stBadOpcode --test=invalidAddr_d31g0v0_Cancun`

Fix:

https://github.com/ethereumjs/ethereumjs-monorepo/pull/3568/commits/2e0f9c5a52f56a0a3e2078e1f0c76279c820af80

**Comment by jochem-brouwer** [comment]
Passes https://github.com/ethereum/execution-spec-tests/releases/tag/eip7692%40v2.1.0, ready for merge :)

File: packages/evm/src/opcodes/functions.ts [hunk 1]
```diff
@@ -1518,6 +1512,30 @@ export const handlers: Map<number, OpHandler> = new Map([
       runState.stack.push(ret)
     },
   ],
+  // 0xf7: RETURNDATALOAD
+  [
+    0xf7,
+    function (runState, _common) {
```
Comment (acolytec3) [inline packages/evm/src/opcodes/functions.ts:1518]:
```suggestion
    function (runState) {
```

**Review by acolytec3 — COMMENTED** [review]
LGTM.  One nit for code cleanliness.

**Review by acolytec3 — DISMISSED** [review]

File: packages/evm/test/eips/eof-header-validation.ts [hunk 2]
```diff
@@ -52,8 +52,8 @@ await new Promise<void>((resolve, reject) => {
 
             const code = hexToBytes(test.code)
 
-            const expected = test.results.Prague.result
-            const _exception = test.results.Prague.exception
+            const expected = test.results.Osaka.result
```
Comment (scorbajio) [inline packages/evm/test/eips/eof-header-validation.ts:55]: It seems like these two added variables aren't used anywhere?

**Review by acolytec3 — APPROVED** [review]

**Pull request merged**